### DISCUSSION TITLE
Civitai Copy/Paste Generation Data Support

### DISCUSSION THREAD

**Discussion by nathanmlim** [Kilvoctu/aiyabot#142] [Ideas]
Would be nice if we could directly copy and paste the generation data directly from civitai directly into the /draw command. There are certain parameters which differ which require manual modifications:

- Example: Civitai --> Aiyabot
- Negative prompt --> negative_prompt
- CFG scale --> guidance_scale
- Model --> data_model

Is this because discord does not allow spaces?

**Comment by Kilvoctu** [comment]
Yeah, no spaces for slash command options.  

I remember recently (when doing [context menu](https://github.com/Kilvoctu/aiyabot/pull/132)) playing with pasting parameters from civitai actually. There was some roundabout way to do so, I'll try to remember what I did.

**Reply by Kilvoctu** [reply]
All right, here's what I did. It's more steps than clicking the `i` but it works 😅. The images on civitai still have their png info in them. I'll go through an example process.  

I'm on the [Solo Levelling style LoRA](https://civitai.com/models/14839) and let's say I like the 2nd preview.  
I click on that image to open it up in its [own page](https://civitai.com/gallery/187735?modelId=14839&modelVersionId=18241&infinite=false&returnUrl=%2Fmodels%2F14839).  
Now I open context menu of image and "Copy Image". For some reason this only works when on the image's own page, and also "Copy Image Link" does not work.   
![image](https://user-images.githubusercontent.com/2993060/223570113-324d61d1-652a-4029-a00c-73aee66f93bb.png)

Then using /identify, drop it into **init_image**, set **phrasing** to "Image Info", and AIYA will read what she can. This is what I got for that image.   
<details>
<summary>dropdown</summary>
<br>

![image](https://user-images.githubusercontent.com/2993060/223570902-d4d69b14-270c-4611-8f98-02848f5f7e4d.png)

</details>

Alternatively, you can save the image, then drop the file into Discord. Afterward, use context menu **Apps** -> **Get Image Info** on the image, and it'll achieve the same result.

**Comment by nathanmlim** [comment]
Appreciate the reply! That's a bummer about no spaces but kinda makes sense. This was an idea for convenience I often find myself using the bot on mobile, so copy pasting the info from the civitai was sort of a pain when trying to fix the spaces on mobile keyboard. 

I wonder if there's a way to add in a separate slash command like /civ that can parse the whole string and pass this into the proper parameters. Might try it this weekend

**Reply by Kilvoctu** [reply]
I tried it on my mobile just now, and I see the process doesn't seem to work 🤔. Even saving the image, it doesn't seem to save the correct one (the one with the png info).
